shared/charts/kube-state-metrics/templates/deployment.yaml

```diff
@@ -0,0 +1,64 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "kube-state-metrics.fullname" . }}
+  labels:
+    app.kubernetes.io/name: {{ include "kube-state-metrics.name" . }}
+    helm.sh/chart: {{ include "kube-state-metrics.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "kube-state-metrics.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "kube-state-metrics.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
+      serviceAccountName: {{ include "kube-state-metrics.fullname" . }}
+      containers:
+      - image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        args:
+          - '--collectors=configmaps,horizontalpodautoscalers,namespaces,pods,ingresses,limitranges,nodes,certificatesigningrequests,cronjobs,deployments,persistentvolumes,jobs,secrets,statefulsets,persistentvolumeclaims,services,storageclasses,poddisruptionbudgets,volumeattachments,daemonsets,endpoints,networkpolicies,replicasets,replicationcontrollers,resourcequotas'
```

Comment:
Two thinks here related to the costs of having custom metrics:

- Are all of those needed? (I've read the line and I think so)
- Would it be better to have this in a variable? is it worth it?

Reply:
> Are all of those needed?

It depends on what metrics are we interested in. I've reviewed the cons of SD metrics and it doesn't seem to be too high (at least compared to logging cost for example), and of course these will only be collected if there are corresponding resources to monitor.

I would leave all of these on initially (there are more, but some not compatible with older K8s API) and once we learn which are useful and which are not, we can reduce this list of collected resources. It's further possible to whitelist only specific namespaces.

> Would it be better to have this in a variable? is it worth it?

Sounds like a good idea, done in the latest commit.